location_client_example.ts

```diff
@@ -14,35 +10,92 @@ async function main() {
     console.log("Create an api for location kind")
     const location_api = kind_crud_api("http://papiea:3000/services", "location_provider", "Location", "0.0.3")
 
-    console.log("\nCreate a new location...")
-    const [metadata] = await location_api.create({ x: 10, y: 20 })
-    console.log("\tCreated location metadata", metadata)
+    {
+        console.log("\nCreate a new location...")
+        const { metadata } = await location_api.create({ x: 10, y: 20 })
+        console.log("\tCreated location metadata", metadata)
+
+        let entity = await location_api.get(metadata)
+        console.log("\tLocation created is:", entity)
+
+        console.log("\nUpdate location to a new spec")
+        await location_api.update(metadata, { x: 100, y: 200 })
```

Comment:
How about the following with immutable object:

```javascript
let loc1: Entity
loc1 = await location_client.create({x: 10, y:20}) // I like the client vs server distinction.. maybe client vs provider.. something like that would be good. "api" and "sdk" are too generic.
loc1 = await loc1.update({x: 11, y:21})
loc1 = loc1.get()
loc1.wait() // Can be called anytime and it waits for the pec in loc1 to be met at some point.
loc1 = loc1.delete() // Returns the object. Deletion can be async as well. Get after delete should work and should return status deleted.. or something like that.
```

Note, there is no need for different types of entities.(Full vs spec only).
The entity ref is not needed if you are creating an entity from scratch.
Otherwise, we need a ref, but that EntityRef should not be "metadata", but only have the three fields we have today in entity references.

```javascript
const lr1: EntityRef = { kind: "location", uuid: "1231-1231-.."}
loc1 = location_client.resolve(lr1)
```

Reply:
Regarding the all purpose `wait`, will get to it when intentful and tasks are in place.

Regarding the different entities, I think my naming was misleading, see my code update.

Regarding `resolve`, its the same is `get` so it is already here. See my next commit.